sgv2-service-common/src/main/java/io/stargate/sgv2/common/grpc/StargateBridgeClientFactory.java

```diff
@@ -16,14 +16,18 @@
 package io.stargate.sgv2.common.grpc;
 
 import io.grpc.Channel;
+import io.stargate.proto.Schema.SchemaRead;
 import java.util.Optional;
 import java.util.concurrent.ScheduledExecutorService;
 
 public interface StargateBridgeClientFactory {
 
   static StargateBridgeClientFactory newInstance(
-      Channel channel, String adminToken, ScheduledExecutorService executor) {
-    return new DefaultStargateBridgeClientFactory(channel, adminToken, executor);
+      Channel channel,
+      String adminToken,
+      SchemaRead.SourceApi sourceApi,
```

Comment:
Something I had missed in the previous PR. `REST` was hard-coded in the common code.

Reply:
Good catch then.

One relatively immediate challenge to consider, however: how do make this extensible for external APIs/Services when we no longer control set of APIs? Should there be some sort of "other" value for someone developing new ones, for example.